cluster/kubernetes/kubernetes.go

```diff
@@ -315,20 +310,21 @@ func (c *Cluster) PublicSSHKey(regenerate bool) (ssh.PublicKey, error) {
 // instance, in which case it returns a list containing the namespaces from the whitelist
 // that exist in the cluster.
 func (c *Cluster) getAllowedNamespaces() ([]apiv1.Namespace, error) {
-	nsList := []apiv1.Namespace{}
+	if len(c.nsWhitelist) > 0 {
+		nsList := []apiv1.Namespace{}
+		for _, name := range c.nsWhitelist {
+			if ns, err := c.client.CoreV1().Namespaces().Get(name, meta_v1.GetOptions{}); err == nil {
+				nsList = append(nsList, *ns)
+			} else if !(apierrors.IsNotFound(err) || apierrors.IsUnauthorized(err) || apierrors.IsForbidden(err)) {
```

Comment:
Do we really want to be silent here? Won't that make it rather hard for users to diagnose why a namespace they specified in their whitelist is getting ignored by flux?

Reply:
It's a tricky one. If the namespace doesn't exist (yet), we'll get either NotFound (if the permissions allow listing namespaces) or Forbidden (if they don't). Failing the whole operation, which underlies much of the API, seems a bit punitive.

We could log it as a warning, I suppose -- that way at least it's flagged up to anyone investigating problems.

Reply:
Now logs it -- could be pretty verbose, but no straight-forward way to avoid that.